server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchemaMonitor.java

```diff
@@ -0,0 +1,233 @@
+            if (!fullSchemaText.equals(currSchemaText.get()))
+            {
+                LOGGER.info("Schema change detected, refreshing CDC tables");
+                currSchemaText.set(fullSchemaText);
+                Set<CqlTable> updatedCdcTables = buildCdcTables(fullSchemaText, databaseAccessor, tableIdCache, instanceFetcher, cassandraBridge);
+                LOGGER.info("Cdc enabled tables tables='{}'", 
+                            updatedCdcTables.stream()
+                                            .map(m -> String.format("%s.%s", m.keyspace(), m.table()))
+                                            .collect(Collectors.joining(",")));
+                cdcTables.set(updatedCdcTables);
+
+                cdcBridge.updateCdcSchema(updatedCdcTables, getPartitioner(nodeSettings),
+                                          ((keyspace, table) -> tableIdCache.get(TableIdentifier.of(keyspace, table))));
+                schemaChangeListeners.forEach(Runnable::run);
+            }
+        }
+        catch (IllegalStateException exception)
+        {
+            LOGGER.warn("There was a problem refreshing the schema. Database Accessor may not be ready", exception);
+            throw exception;
+        }
+        catch (Throwable t)
+        {
+            LOGGER.error("Unexpected error while refreshing the schema", t);
+            throw t;
+        }
+    }
+
+    private Partitioner getPartitioner(NodeSettings nodeSettings)
+    {
+        if (nodeSettings.partitioner().contains("."))
+        {
+            String[] splitPartitionerName = nodeSettings.partitioner().split("\\.");
+            return Partitioner.valueOf(splitPartitionerName[splitPartitionerName.length - 1]);
+        }
+        return Partitioner.valueOf(nodeSettings.partitioner());
+    }
+
+    @Override
+    public DurationSpec delay()
+    {
+        return sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime();
+    }
+
+    @Override
+    public void execute(Promise<Void> promise)
+    {
+        try
+        {
+            refresh();
+            promise.tryComplete();
+        }
+        catch (Throwable t)
+        {
+            promise.fail(t);
+        }
+    }
+
+    @Override
+    public ScheduleDecision scheduleDecision()
+    {
+        if (sidecarConfiguration.serviceConfiguration().schemaKeyspaceConfiguration().isEnabled() &&
+            sidecarConfiguration.serviceConfiguration().cdcConfiguration().isEnabled())
+        {
+            return ScheduleDecision.EXECUTE;
+        }
+        return ScheduleDecision.SKIP;
+    }
+
+    @VisibleForTesting
+    static Set<CqlTable> buildCdcTables(CdcDatabaseAccessor cdcDatabaseAccessor,
+                                        ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                        @NotNull InstanceMetadataFetcher instanceFetcher,
+                                        @NotNull final CassandraBridge cassandraBridge)
```

Comment:
instanceFetcher is no longer needed
```suggestion
    static Set<CqlTable> buildCdcTables(CdcDatabaseAccessor cdcDatabaseAccessor,
                                        ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
                                        @NotNull CassandraBridge cassandraBridge)
```